xcdat/dataset.py

```diff
@@ -100,7 +100,10 @@ def open_dataset(
     ds = xr.open_dataset(path, decode_times=False, **kwargs)  # type: ignore
 
     if decode_times:
-        ds = decode_time(ds)
+        try:
+            ds = decode_time(ds)
+        except KeyError as e:
+            logger.warning(e)
```

Comment:
This is the key fix – if `decode_time` fails then `ds` remains as is and a warning is printed. No need to update the `decode_times` boolean. This makes sense to me.